modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -151,6 +151,243 @@ GAPI_FLUID_KERNEL(GFluidAddW, cv::gapi::core::GAddW, false)
 
 enum Arithm { ARITHM_ABSDIFF, ARITHM_ADD, ARITHM_SUBTRACT, ARITHM_MULTIPLY, ARITHM_DIVIDE };
 
+#if CV_SIMD
+template< typename VT, typename T>
+static inline void absdiff_impl(const T in1[], const T in2[], T out[], int length, int& x)
+{
+    constexpr int nlanes = VT::nlanes;
+
+    for (; x < length; )
+    {
+        for (; x <= length - nlanes; x += nlanes)
+        {
+            VT a = vx_load(&in1[x]);
+            VT b = vx_load(&in2[x]);
+            vx_store(&out[x], v_absdiffs(a, b));
+        }
+
+        if (x < length && length >= nlanes)
+        {
+            x = length - nlanes;
+        }
+    }
+}
+
+template<typename T>
+static inline void absdiff_simd(const T in1[], const T in2[], T out[], int length, int& x)
+{
+    if (std::is_same<T, uchar>::value)
+    {
+        absdiff_impl<v_uint8>(reinterpret_cast<const uchar*>(in1),
+                                     reinterpret_cast<const uchar*>(in2),
+                                     reinterpret_cast<uchar*>(out), length, x);
+    }
+    else if (std::is_same<T, ushort>::value)
+    {
+        absdiff_impl<v_uint16>(reinterpret_cast<const ushort*>(in1),
+                                       reinterpret_cast<const ushort*>(in2),
+                                       reinterpret_cast<ushort*>(out), length, x);
+
+    }
+    else if (std::is_same<T, short>::value)
+    {
+        absdiff_impl<v_int16>(reinterpret_cast<const short*>(in1),
+                                     reinterpret_cast<const short*>(in2),
+                                     reinterpret_cast<short*>(out), length, x);
+    }
+    else if (std::is_same<T, float>::value)
+    {
+        absdiff_impl<float, v_float32>(reinterpret_cast<const float*>(in1),
+                                       reinterpret_cast<const float*>(in2),
+                                       reinterpret_cast<float*>(out), length, x);
+    }
+}
```

Comment:
I'd move `absdiff_simd` to overload mechanism here: 
```
// Default implementation
template<typename T>
static inline void absdiff_simd(const T in1[], const T in2[], char T[], int length, int& x) {
    // Do nothing
}

// Special implementation which will prevale over the default one
static inline void absdiff_simd(const char in1[], const char in2[], char out[], int length, int& x) {
    absdiff_impl<v_uint8>(in1, in2, out, length, x);
}
```
Also you can add a traits not to declare every overloaded function. Or you can file a ticket on that :)

Reply:
I've already implemented this feature via overloads of "..._simd" functions. However, this didn't provide brevity, beauty of snippet or performance gains. It also gives a lot of nested calls which doesn't speed up performance. So I'd prefer to leave this implementation as is. :-)